src/Symfony/Component/Cache/Traits/FilesystemTrait.php

```diff
@@ -23,6 +23,33 @@
     use FilesystemCommonTrait;
 
     /**
+     * @param int|null $time
+     *
+     * @return int
+     */
+    public function prune($time = null)
+    {
+        $time = $time ?: time();
+        $i = 0;
+
+        foreach (new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($this->directory, \FilesystemIterator::SKIP_DOTS), \RecursiveIteratorIterator::CHILD_FIRST, \RecursiveIteratorIterator::CATCH_GET_CHILD) as $file) {
+            // remove empty directories
+            if ($file->isDir() && 2 === count(scandir($file->getPathname()))) {
+                @rmdir(rtrim($file->getPathname(), '/.'));
+
+                continue;
+            }
+
+            // remove items older than passed (or current) time
+            if ($time >= $file->getMTime() && @unlink($file) && !file_exists($file)) {
+                ++$i;
+            }
+        }
+
+        return $i;
```

Comment:
I've gone ahead and changed the implementation to return `bool` for now; please advise if you find value in the above and I can revert; for now, it's as requested!